### PR TITLE
Make the use of @Blocking and @NonBlocking a build time error

### PR DESCRIPTION
**Pull request by geoand**
This was initially brought up in
https://github.com/quarkusio/quarkus/pull/19585#discussion_r694077653

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/BothBlockingAndNonBlockingTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void test() {
+        fail("Should never have been called");
+    }
+
+    @Path("test")
+    public static class Resource {
+
+        @Path("hello")
+        @Blocking
```
Comment (cescoffier) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/BothBlockingAndNonBlockingTest.java:40]: Should we also test on classes and application classes?

**Review by cescoffier — APPROVED** [review]
Should we add tests for classes and application classes?

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/BothBlockingAndNonBlockingTest.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/BothBlockingAndNonBlockingTest.java:40]: Sure, I'll add those tests

**Comment by michalszynkiewicz** [comment]
@geoand no way we could make this global?

**Comment by geoand** [comment]
> @geoand no way we could make this global?

Global? How do you mean?

**Comment by michalszynkiewicz** [comment]
> > @geoand no way we could make this global?
> 
> Global? How do you mean?

Not in RestEasy Reactive but in some core module.

Having `@Blocking` and `@NonBlocking` on a class or method anywhere is an error, not only on REST or gRPC, no?

**Comment by geoand** [comment]
I am not sure we want that TBH. It's not something we support in RESTEasy Reactive yet. But feel free to send an email to the dev-list

**Comment by michalszynkiewicz** [comment]
Are you saying that you think there may be scenarios in which a class like:
```
@NonBlocking
@Blocking
public class Foo {
}
```

or a method like this:

```java
@NonBlocking
@Blocking
public void foo() {
}
```
is valid?

Or that it's not clear where we could add a validation for this?

**Comment by geoand** [comment]
In RESTEasy Reactive, that will not work and I am pretty sure we don't want it to work. 
But I don't think it makes sense to check it as we don't check for known annotations in random places in either extensions

**Comment by cescoffier** [comment]
No, it should not work and be reported to the user as invalid.

**Comment by geoand** [comment]
But that would mean that in all our extensions we would do this kind of check... I don't think it makes too much sense and I don't see a case where users would just think they can just an annotation anywhere...

**Comment by michalszynkiewicz** [comment]
Why every extension? A processor in quarkus-core-deployment that would do the check on all user's classes would suffice, no?

**Comment by geoand** [comment]
What I'm saying is that extensions usually don't check for every case of invalid annotation usage - only the most common

**Pull request merged**